plan/physical_plan_builder.go

```diff
@@ -179,23 +185,39 @@ func (ds *DataSource) findBestTask(prop *requiredProp) (task, error) {
 	if t != nil {
 		return t, nil
 	}
+
+	oldPropCols := prop.cols
```

Comment:
Why we need this?

Reply:
Because we need to set `prop.cols = nil` in `DataSource.findBestTask`when `prop.enforced == true`. And reset it for storing `map<prop, task>` before the function return.

Reply:
It's better to add a comment here.

Reply:
Already add.